### PR TITLE
Correct maximum filename length value (fixes #3338)

### LINKED ISSUE #3338: EPIC: Syncthing 0.13 doesn't sync _some_ files
### Version Information

Syncthing Version: v0.13.7
OS Version: Ubuntu 14.04 (also affects Windows, MacOS, etc)

### Steps to reproduce the problem

This bug can be reproduced in different environments. Please don't get an impression that only Linux users are affected.

Here is an example how to reproduce an issue:

1. Install Ununtu 14.04 LTS, enable home directory encryption using EncFS. EncFS has a serious limit on maximum file name length which will help us to reproduce an issue.
2. Install Syncthing
3. Create a file with long enough name (close to FS limits) in Sync directory, for instance: `/home/eax/Sync/books/by_topic/papers-redbook/0032 - ARIES- A Transaction Recovery Method Supporting Fine-Granularity Locking and Partial Rollbacks Using Write-Ahead Logging.pdf`
4. Sync with second machine that doesn't use EncFS
5. Delete Syncthing, clear Sync directory
6. Re-Install Syncthing, try to sync with second machine

### What you expected to happen

All files sync successfully.

### What actually happened

Syncthing doesn't sync some files. In log there are multiple error messages like this:

```
[TQISY] 10:29:46 INFO: Puller: final: open /home/eax/Sync/books/by_topic/papers-redbook/.syncthing.032 - ARIES- A Transaction Recovery Method Supporting Fine-Granularity Locking and Partial Rollbacks Using Write-Ahead Logging.pdf.tmp: file name too long 
```

### Suggestion

Temp file name is longer than original file name and thus doesn't satisfy filesystem limits. Perhaps instead of `.syncthing.(full file name).tmp` you could use `.syncthing.(sha256 of file name).tmp`.

### PR DESCRIPTION
**Pull request by afiskon**
See discussion https://github.com/syncthing/syncthing/issues/3338

### CODE REVIEW

**Comment by AudriusButkevicius** [comment]
So it still doesn't explain the fact that the file that was failing for you was around 140 characters, nowhere close 179

**Comment by calmh** [comment]
Better yet, does this *work* for you with encfs?

**Comment by afiskon** [comment]
@AudriusButkevicius you should consider a **full** filename:

```
$ python3

>>> len("""/home/eax/Sync/books/by_topic/papers-redbook/.syncthing.032 - ARIES- A Transaction Recovery Method Supporting Fine-Granularity Locking and Partial Rollbacks Using Write-Ahead Logging.pdf.tmp""")

190
```

**Comment by calmh** [comment]
@st-jenkins add to whitelist

**Comment by calmh** [comment]
Wait what? Why should we consider that? :)

File: lib/model/tempname.go [hunk 1]
```diff
@@ -20,6 +20,13 @@ type tempNamer struct {
 
 var defTempNamer tempNamer
 
+// (max filename length supported by FS) - len(".syncthing.") - len(".tmp")
+// Worst case is EncFS which according to man page supports filenames up to
+// "approximately" 3*(N-2)/4 characters where N is a limit of underlying FS.
+// If underlying FS is ext4 in practice it means that maximum filename length
+// is 188 characters. Minus len(".syncthing."), minus len(".tmp") gives:
+const maxFilenameLength = 173
```
Comment (calmh) [inline lib/model/tempname.go]: I'd make this

```go
const maxFilenameLength = 175 - len(".syncthing.") - len(".tmp")
```

to match observed reality. The "approximately" was very approximate indeed in my test case above.

**Pull request closed without merging**